crates/ruff_db/src/vendored.rs

```diff
@@ -0,0 +1,379 @@
+use std::cell::{RefCell, RefMut};
+use std::io::{self, Read};
+use std::ops::{Deref, DerefMut};
+use std::sync::{Mutex, MutexGuard};
+
+use itertools::Itertools;
+use once_cell::sync::Lazy;
+use zip::{read::ZipFile, ZipArchive, ZipWriter};
+
+pub use path::{VendoredPath, VendoredPathBuf};
+
+pub mod path;
+
+type Result<T> = io::Result<T>;
+
+#[derive(Debug)]
+pub struct VendoredFileSystem {
```

Comment:
Nit: I would just call this `Vendored` or `VendoredFiles` to avoid any confusion with `FileSystem`.